eng/common/scripts/copy-docs-to-blobstorage.ps1

```diff
@@ -196,7 +209,24 @@ function Upload-Blobs
     Write-Host "DocVersion $($DocVersion)"
     Write-Host "DocDir $($DocDir)"
     Write-Host "Final Dest $($DocDest)/$($PkgName)/$($DocVersion)"
+    Write-Host "Release Tag $($ReleaseTag)"
 
+    # Use the step to replace master link to release tag link 
+    if ($ReleaseTag) {
+        foreach ($htmlFile in (Get-ChildItem $DocDir -include *.html -r)) 
+        {
+            $fileContent = Get-Content -Path $htmlFile
+            $checkPattern = $false
+            while ($fileContent -match $RepoReplaceRegex) {
```

Comment:
It is probably cheaper to always do the replace and then do string comparison to check if something changes instead of doing the regex call twice (once for match and once for replace).

Reply:
The pain point is if two matches in one line, then -replace only can replace one of them. So we have to do a while loop check until no matches.

Reply:
Are you sure about that I just did a simple test locally and it replaced all instances of it for me. I think the issue is with the regex being used is too greedy with the `.*` at the end which causes it to consume all the other chars after. Instead we should do something like `[^ ")]*` which will match all chars except space, quote and paren (we can add other chars if we find they should match the end). 

This will be more efficient then looping over the file contents multiple times and replacing only one each time.